Movecraft/src/main/java/net/countercraft/movecraft/processing/tasks/detection/DetectionTask.java

```diff
@@ -339,35 +352,79 @@ private DetectAction(ConcurrentLinkedQueue<MovecraftLocation> currentFrontier, C
         @Override
         public void run() {
             MovecraftLocation probe;
+
             while((probe = currentFrontier.poll()) != null) {
-                visitedMaterials.computeIfAbsent(movecraftWorld.getMaterial(probe), Functions.forSupplier(ConcurrentLinkedDeque::new)).add(probe);
+                BlockData bd = movecraftWorld.getData(probe);
+                Material m = bd.getMaterial();
+                boolean blockFacingCraft = true;
+
+                if (m == Material.LADDER) {
```

Comment:
Consider using pattern matched switch statements by increasing the java version to java 21 - it is a requirement for Minecraft 1.20+

Reply:
1.18.2 only requires Java 17, so we can't use Java 21 features until we drop support for it.

Reply:
why not?

Reply:
I can add a TODO when ready to update to java 21

Reply:
> why not?
Paper (and Spigot) will not start up with Java 21 unless the server owner explicitly bypasses their check.  While I do so on our servers, I do not plan on asking all server owners running Movecraft to do so.